packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,520 @@
+
+                return {
+                    id: palette.id,
+                    title: palette.name,
+                    colors,
+                };
+            })
+        );
+    }, [appBridgePalettes, appBridge]);
+
+    useEffect(() => {
+        if (blockSettings?.colorInput) {
+            setDisplayableItems(blockSettings.colorInput);
+        }
+        const currentHeight = blockSettings.customHeight ? blockSettings.heightInput : blockSettings.heightSlider;
+        if (colorScaleHeight !== currentHeight) {
+            setColorScaleHeight(currentHeight);
+        }
+        if (colorScaleBlockRef?.current) {
+            const minimumColors = getMinimumAmountOfSquares(blockSettings.colorInput);
+            if (blockSettings.colorInput && blockSettings.colorInput.length >= minimumColors) {
+                const needToCalculateWidths = blockSettings.colorInput.find((color) => !color.width);
+                if (needToCalculateWidths && blockSettings.colorInput.length > 0) {
+                    const colorsWithNewWidths = calculateWidths(blockSettings.colorInput, colorScaleBlockRef, false);
+                    setBlockSettings({
+                        ...blockSettings,
+                        colorInput: colorsWithNewWidths,
+                    });
+                }
+            }
+        }
+    }, [blockSettings, colorScaleHeight, setBlockSettings]);
+
+    const handleResizeStop = () => {
+        draggingId.current = null;
+    };
+
+    const handleResizeStart = (event: MouseEvent, index?: number, currentColor?: ColorProps): void => {
+        if (dragStartPos) {
+            dragStartPos.current = event.clientX;
+            dragStartWidth.current = currentColor?.width;
+        }
+        draggingId.current = index;
+    };
+
+    const handleResize = (event: MouseEvent) => {
+        if (draggingId.current !== null) {
+            const id = draggingId.current;
+            if (!lastDragPos.current) {
+                lastDragPos.current = event.clientX;
+            }
+
+            if (event.clientX < lastDragPos?.current && lastDragPos.current - event.clientX >= 8) {
+                lastDragPos.current = event.clientX;
+
+                let valuesChanged = false;
+
+                if (!dragStartPos.current) {
+                    return;
+                }
+
+                const movementSinceStart = dragStartPos.current - event.clientX;
+
+                const colorsBeforeCurrent = displayableItems
+                    ?.filter((_, diIndex) => id !== undefined && diIndex < id)
+                    .reverse();
+
+                const newDisplayableItems = displayableItems?.map((diValue, diIndex) => {
+                    if (diValue && diIndex === id) {
+                        if (diValue.width && diValue.width >= 16) {
+                            // need to make sure it's 16 because we're going to decrease width
+                            // by 8 pixels, and the minimum width is 8 pixels
```

Comment:
then define 8 as OFFSET_WIDTH and the other 8 define as MINIMUM_WIDTH. Instead of 16 use MINIMUM_WIDTH * 2